soccer/radio/PacketConvert.cpp

```diff
@@ -0,0 +1,78 @@
+#include <Geometry2d/Util.hpp>
+#include <iostream>
+#include <time.hpp>
+#include <status.h>
+#include "PacketConvert.hpp"
+
+void from_robot_tx_proto(const Packet::Robot& proto_packet, rtp::RobotTxMessage* msg) {
+    Packet::Control control = proto_packet.control();
+    msg->uid = proto_packet.uid();
+    msg->message.controlMessage.bodyX = static_cast<int16_t>(
+        control.xvelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR);
+    msg->message.controlMessage.bodyY = static_cast<int16_t>(
+        control.yvelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR);
+    msg->message.controlMessage.bodyW = static_cast<int16_t>(
+        control.avelocity() * rtp::ControlMessage::VELOCITY_SCALE_FACTOR);
+    msg->message.controlMessage.dribbler =
+        clamp(static_cast<uint16_t>(control.dvelocity()) * 2, 0, 255);
+
+    msg->message.controlMessage.shootMode = control.shootmode();
+    msg->message.controlMessage.kickStrength = control.kcstrength();
+    msg->message.controlMessage.triggerMode = control.triggermode();
+    msg->message.controlMessage.song = control.song();
+    msg->messageType = rtp::RobotTxMessage::ControlMessageType;
+}
+
+void convert_tx_proto_to_rtp(const Packet::RadioTx &proto_packet,
+                             rtp::RobotTxMessage *messages) {
+    from_robot_tx_proto(proto_packet.robots(0), messages);
```

Comment:
I broke this compared to the original pr so that each robot only gets it's own data

Reply:
I thought that's how I originally had it?

Reply:
Oh right. I had one implementation that filled them all in, and then I looped over them to send. This should be fine, then.